aiida/orm/computers.py

```diff
@@ -463,11 +477,28 @@ def set_default_mpiprocs_per_machine(self, def_cpus_per_machine: Optional[int])
         """
         if def_cpus_per_machine is None:
             self.delete_property('default_mpiprocs_per_machine', raise_exception=False)
-        else:
-            if not isinstance(def_cpus_per_machine, int):
-                raise TypeError('def_cpus_per_machine must be an integer (or None)')
+        elif not isinstance(def_cpus_per_machine, int):
+            raise TypeError('def_cpus_per_machine must be an integer (or None)')
         self.set_property('default_mpiprocs_per_machine', def_cpus_per_machine)
 
+    def get_default_memory_per_machine(self) -> Optional[int]:
+        """
+        Return the default amount of memory (kB) per machine (node) for this computer,
+        or None if it was not set.
+        """
+        return self.get_property('default_memory_per_machine', None)
+
+    def set_default_memory_per_machine(self, def_memory_per_machine: Optional[int]) -> None:
+        """
+        Set the default amount of memory (kB) per machine (node) for this computer.
+        Accepts None if you do not want to set this value.
+        """
+        if def_memory_per_machine is None:
+            self.delete_property('default_memory_per_machine', raise_exception=False)
```

Comment:
I realize that other methods do this, but in this case explicitly deleting the property when setting `None` doesn't really make sense. Just setting the thing to `None` should also work no? And when you retrieve it, you will get `None`, just as if the property doesn't even exist.

Reply:
I propose to do this in the next PR. I am just going to add an item to fix this for other methods in https://github.com/aiidateam/aiida-core/issues/5257

Reply:
Also, just to mention that the presence of this line requires me to put the line in the tests that you've suggested to remove in https://github.com/aiidateam/aiida-core/pull/5260/files#r766608219

Reply:
Why exactly? If you don't set it, it should be `None`, which is an allowed value is it? What test is failing where exactly when you remove the explicit setting to `0`?

Reply:
> Why exactly? If you don't set it, it should be `None`, which is an allowed value is it?

For the `def_memory_per_machine` attribute (and for many other attributes) setting to `None` makes them be deleted. That was, apparently, the convention before. I would like to change that as the response to #5257. I think if an attribute is set to `None`, it should remain just `None` and not be deleted.

> What test is failing where exactly when you remove the explicit setting to `0`?

those:
```
tests/cmdline/commands/test_computer.py::TestVerdiComputerConfigure::test_local_from_config FAILED                                                                                                   [ 47%]
tests/cmdline/commands/test_computer.py::TestVerdiComputerConfigure::test_local_interactive FAILED                                                                                                   [ 50%]
tests/cmdline/commands/test_computer.py::TestVerdiComputerConfigure::test_local_ni_empty FAILED                                                                                                      [ 52%]
tests/cmdline/commands/test_computer.py::TestVerdiComputerConfigure::test_show FAILED                                                                                                                [ 58%]
tests/cmdline/commands/test_computer.py::TestVerdiComputerConfigure::test_ssh_interactive FAILED                                                                                                     [ 61%]
tests/cmdline/commands/test_computer.py::TestVerdiComputerConfigure::test_ssh_ni_empty FAILED                                                                                                        [ 64%]
tests/cmdline/commands/test_computer.py::TestVerdiComputerConfigure::test_ssh_ni_username FAILED
```

Reply:
Just ran the tests and so it is the `ComputerBuilder` that complains, but it is wrong to. The `default_memory_per_machine` is optional on the ORM level (and so also on the CLI) and so it shouldn't fail if it is not explicitly specified. But ok, since we agreed that the builder would be changed/removed in #5257 we can leave this as is for now. But I really think we should get rid of that thing a.s.a.p.